packages/medusa/src/services/line-item.ts

```diff
@@ -378,6 +378,31 @@ class LineItemService extends BaseService {
     )
   }
 
+  /**
+   * Deletes a line item with the tax lines.
+   * @param {string} id - the id of the line item to delete
+   * @return {Promise<LineItem | undefined>} the result of the delete operation
+   */
+  async deleteWithTaxLines(id: string): Promise<LineItem | undefined> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        const lineItemRepository = transactionManager.getCustomRepository(
+          this.lineItemRepository_
+        )
+
+        const itemTaxLineRepo = transactionManager.getCustomRepository(
```

Comment:
todo: We should not use the repo directly here. instead we should probably use `taxProviderServiceTs.clearLineItemsTaxLines`

Reply:
+1